### PR TITLE
Console

### PR DESCRIPTION
**Pull request by aztech-dev**
- Keypair generation in PEM format
- Public key from PEM private key
- Dump key info for pub & priv keys
- Helper commands (hex encode / decode)

All PEM related commands are compatible with openssl generated EC keys (though they can't parse the data as is, BEGIN/END tags must be removed, and new lines stripped)

### CODE REVIEW

**Comment by btcdrak** [comment]
Beautiful.

File: composer.json [hunk 1]
```diff
@@ -44,5 +47,11 @@
       "psr-4":{
          "Mdanter\\Ecc\\Tests\\":"tests/unit"
       }
-   }
+   },
+   "repositories": [
+        {
+            "type": "vcs",
+            "url": "git@github.com:aztech-digital/PHPASN1.git"
```
Comment (aztech-dev) [inline composer.json]: Library is forked from https://github.com/FGrosse/PHPASN1, but it currently lacks composer support. I've issued a P/R to fix that, so if/when it does get fixed and the owner registers the package in packagist, we can remove this (or point it to the source repo if he merges but does not add to packagist)

File: composer.json [hunk 2]
```diff
@@ -25,11 +25,14 @@
       }
    ],
    "require":{
-       "php": ">=5.3.3"
+       "php": ">=5.3.3",
+       "fgrosse/phpasn1": "dev-master",
```
Comment (btcdrak) [inline composer.json]: I dont like tagging to master because it means the lib might break as they develop. We need to use tags.

**Comment by aztech-dev** [comment]
Thanks. Next step is message signing/verification but I'm short on time for this week.

File: composer.json (hunk 2 above)
Comment (aztech-dev) [inline composer.json]: Agreed, I think we can branch alias or ref a specific commit. But given the level of activity on the repo, I dont think there is much risk (he also uses a develop branch). Anyways, I'm waiting for feedback on the P/R I issued to improve composer support.

**Comment by btcdrak** [comment]
I can merge this now if you would like.

**Comment by aztech-dev** [comment]
Go ahead, tests do pass (the last two commit have no effect)

**Comment by rubensayshi** [comment]
awesome!

**Comment by aztech-dev** [comment]
Yup, except no one noticed (me included) that I had left the bin/ directory in .gitignore, and therefore, there's no script to run the commands :D

**Pull request merged**